etl/steps/data/meadow/un_sdg/2022-05-26/un_sdg.py

```diff
@@ -0,0 +1,404 @@
+    # Load and clean the data
+    log.info("Reading in original data...")
+    original_df = original_df.copy(deep=False)
+
+    # removing values that aren't numeric e.g. Null and N values
+    original_df.dropna(subset=["Value"], inplace=True)
+    original_df.dropna(subset=["TimePeriod"], how="all", inplace=True)
+    original_df = original_df[
+        pd.to_numeric(original_df["Value"], errors="coerce").notnull()
+    ]
+    original_df.rename(
+        columns={"GeoAreaName": "Country", "TimePeriod": "Year"}, inplace=True
+    )
+    return original_df
+
+
+def create_short_unit(long_unit: pd.Series) -> np.ndarray:
+
+    conditions = [
+        (long_unit.str.contains("PERCENT"))
+        | (long_unit.str.contains("Percentage") | (long_unit.str.contains("%"))),
+        (long_unit.str.contains("KG")) | (long_unit.str.contains("Kilograms")),
+        (long_unit.str.contains("USD")) | (long_unit.str.contains("usd")),
+    ]
+
+    choices = ["%", "kg", "$"]
+
+    short_unit = np.select(conditions, choices, default=None)
+    return short_unit
+
+
+def generate_tables_for_indicator_and_series(
+    data_series: pd.DataFrame,
+    init_dimensions: tuple,
+    init_non_dimensions: tuple,
+    dim_dict: dict,
+) -> pd.DataFrame:
+    tables_by_combination = {}
+    data_dimensions, dimensions, dimension_values = get_series_with_relevant_dimensions(
+        data_series, init_dimensions, init_non_dimensions
+    )
+    if len(dimensions) == 0:  # not the best solution.
+        # no additional dimensions
+        export = data_dimensions
+        return export
+    else:
+        dim_desc = (
+            dim_dict.set_index("id")
+            .loc[dimensions]
+            .set_index("code")
+            .squeeze()
+            .to_dict()
+        )
+        dim_desc["nan"] = ""
+        dim_desc["None"] = ""
+        i = 0
+        # Mapping the dimension value codes to more meaningful descriptions
+        for i in range(len(dimension_values)):
+            df = pd.DataFrame({"value": dimension_values[i]})
+            df["value"] = df["value"].astype(str)
+            dimension_values[i] = [dim_desc[k] for k in df["value"].to_list()]
+        # Mapping the descriptions into the dataframe
+        for dim in dimensions:
+            data_dimensions[dim] = data_dimensions[dim].astype(str)
+            data_dimensions[dim] = [dim_desc[k] for k in data_dimensions[dim]]
+        # Create each combination of dimension values, e.g. each age group & sex combination. Not all combinations will have associated data.
+        for dimension_value_combination in itertools.product(*dimension_values):
+            # build filter by reducing, start with a constant True boolean array
+            filt = [True] * len(data_dimensions)
+            for dim_idx, dim_value in enumerate(dimension_value_combination):
+                dimension_name = dimensions[dim_idx]
+                value_is_nan = type(dim_value) == float and math.isnan(dim_value)
```

Comment:
It's recommended to use `isinstance(dim_value, float)` instead of `type(dim_value) == float` but honestly I don't remember why... :)